platform.h

```diff
@@ -65,11 +68,28 @@ typedef struct calvinsys_sensors_environmental_t {
   result_t (*get_temperature)(double *temp);
 } calvinsys_sensors_environmental_t;
 
-void platform_init(struct node_t *node);
+typedef struct platform_t {
+#ifdef PLATFORM_ANDROID
+	int upstream_platform_fd[2]; // read end [0], write end [1]
+	int downstream_platform_fd[2];
+#endif
+} platform_t;
+
+#ifdef PLATFORM_ANDROID
+struct platform_command_handler_t {
+	char command[50];
+	result_t (*handler)(struct node_t* node, char *data, size_t size);
+};
+result_t platform_android_handle_data(struct node_t* node, struct transport_client_t *transport_client);
```

Comment:
Move this definition to platform_android.c and make it static as it is only used there.